<?xml version="1.0"?>
<launch>
    <!-- <master auto="start"/>  -->
    <param name="use_sim_time" value="true" />
    <param name="cfgfile" value="$(find xnrobot_id11)/param/znnz2021.yaml" />

    <!-- Run the map server -->
<!--     <node name="map_server" pkg="map_server" type="map_server" args="$(find xnrobot_id11)/map/map.yaml"/> -->

    <node pkg="xnrobot_id11" type="znnz2021_tf_manager.py" name="znnz2021_tf_manager">
    </node>

    <node pkg="xnrobot_id11" type="znnz2021_filter_laser.py" name="znnz2021_filter_laser">
    </node>

    <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">

        <!-- Frame names -->
        <param name="map_frame" value="map" />
        <param name="base_frame" value="base_link" />
        <param name="odom_frame" value="odom" />

        <!-- Tf use -->
        <param name="use_tf_scan_transformation" value="true"/>
        <param name="use_tf_pose_start_estimate" value="false"/>
        <param name="pub_map_odom_transform" value="true"/>

        <!-- Map size / start point -->
        <param name="map_resolution" value="0.01"/>
        <param name="map_size" value="1024"/>
        <param name="map_multi_res_levels" value="2" />

        <!-- Map update parameters -->
        <param name="update_factor_free" value="0.4"/>
        <param name="update_factor_occupied" value="0.9" />
        <param name="map_update_distance_thresh" value="0.2"/>
        <param name="map_update_angle_thresh" value="0.2" />
        <param name="laser_z_min_value" value = "-1.0" />
        <param name="laser_z_max_value" value = "1.0" />

        <!-- Advertising config -->
        <param name="advertise_map_service" value="true"/>

        <param name="scan_subscriber_queue_size" value="5"/>
        <param name="scan_topic" value="scan_valid"/>

        <param name="tf_map_scanmatch_transform_frame_name" value="scanmatcher_frame" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find xnrobot_id11)/rviz_cfg/znnz2021_amcl.rviz"/>
</launch> 

